matching-core/src/test/java/com/graphhopper/matching/MapMatchingTest.java

```diff
@@ -289,6 +290,81 @@ public void testRepairUTurn() {
         mm.checkOrCleanup(res, false);
     }
 
+    @Test
+    public void testWayGeometryAtStartAndEndPoint() {
+        GraphHopperStorage graph = hopper.getGraphHopperStorage();
+        LocationIndexMatch locationIndex = new LocationIndexMatch(graph,
+                (LocationIndexTree) hopper.getLocationIndex());
+        MapMatching mapMatching = new MapMatching(graph, locationIndex, encoder);
+
+        // https://graphhopper.com/maps/?point=51.341708%2C12.385272&point=51.340622%2C12.388405&locale=de-DE&vehicle=car&weighting=fastest&elevation=true&layer=Omniscale
+        List<GPXEntry> inputGPXEntries = new GPXFile().doImport("./src/test/resources/testStartEndGeometry.gpx").getEntries();
+        MatchResult mr = mapMatching.doWork(inputGPXEntries,true);
+
+        NodeAccess nodeAccess = graph.getNodeAccess();
+
+        List<EdgeMatch> edgeMatches = mr.getEdgeMatches();
+        for (int i = 0; i < edgeMatches.size(); i++) {
+            EdgeMatch match = edgeMatches.get(i);
+            assertNotNull(match.getWayGeometry());
+            if (i == 0) {
+                assertEquals(2, match.getWayGeometry().size());
+                assertEquals(51.341708, match.getWayGeometry().getLat(0), 1E-5);
+                assertEquals(12.385272, match.getWayGeometry().getLon(0), 1E-5);
+                assertEquals(nodeAccess.getLat(mr.getEdgeMatches().get(0).getEdgeState().getAdjNode()), match.getWayGeometry().getLat(1), 1E-5);
+                assertEquals(nodeAccess.getLon(mr.getEdgeMatches().get(0).getEdgeState().getAdjNode()), match.getWayGeometry().getLon(1), 1E-5);
+            } else if (i == edgeMatches.size() - 1) {
+                assertEquals(2, match.getWayGeometry().size());
+                assertEquals(nodeAccess.getLat(mr.getEdgeMatches().get(mr.getEdgeMatches().size() - 1).getEdgeState().getBaseNode()), match.getWayGeometry().getLat(0), 1E-5);
+                assertEquals(nodeAccess.getLon(mr.getEdgeMatches().get(mr.getEdgeMatches().size() - 1).getEdgeState().getBaseNode()), match.getWayGeometry().getLon(0), 1E-5);
+                assertEquals(51.340622, match.getWayGeometry().getLat(1), 1E-5);
+                assertEquals(12.388405, match.getWayGeometry().getLon(1), 1E-5);
+            }
+        }
+
+        assertEquals(mr.getGpxEntriesLength(), mr.getMatchLength(), 1);
+        // TODO why is there such a big difference for millis?
```

Comment:
what do you mean with this TODO and compared to which value this is a 'big' difference?